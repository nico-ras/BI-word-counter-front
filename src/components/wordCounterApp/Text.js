import React from "react";

export const Text = ({ text }) => {
  return (
    <div>
      <h4>Texto analizado: </h4>
      <p data-testid='test-text'>{text}</p>
    </div>
  );
};
