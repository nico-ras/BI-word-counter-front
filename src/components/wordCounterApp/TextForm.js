import React from "react";
import '../../styles/wordCounter/textForm.css'

export const TextForm = ({ textId, onInputChange, getFetch }) => {
  return (
    <div>
      <h2 className="formTitle">Ranking de palabras por texto</h2>

      <h3 className="fromSubtitle">Selecciona el ID del texto que deseas encontrar: </h3>
      <form>
        <input
          className="idInput"
          data-testid="inputId"
          type="number"
          name="textId"
          placeholder="ID"
          value={textId}
          onChange={onInputChange}
        />
        <input className="rankingBtn" type="button" value="Ver ranking" onClick={getFetch}></input>
      </form>
    </div>
  );
};
