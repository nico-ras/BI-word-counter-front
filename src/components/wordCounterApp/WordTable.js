import React from 'react'

export const WordTable = ({element, element:{position, word, occurrences}}) => {
  return (
    <>
       <tr key={element}>
        <td>{position}</td>
        <td>'{word}'</td>
        <td>{occurrences}</td>
        <td>Ver texto</td>
        <td>Editar</td>
       </tr>
    </>
  )
}


