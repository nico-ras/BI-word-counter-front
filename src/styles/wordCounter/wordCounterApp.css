.app {
    margin: auto;
    margin-top: 4em;
    width: 40em;
    display: inline-block;
    padding-bottom: 2em;
    border: solid 1px #D3D3D3;
    border-radius: 0.5em;
    background-color: #FFFFFF;
}

h1 {
    color: #001A72;
    font-size: 3rem;
}

.wcaTitle {
    color: #003669;
}