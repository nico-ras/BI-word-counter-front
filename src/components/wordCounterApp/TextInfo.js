import React, { useState } from "react";
import "../../styles/wordCounter/textInfo.css";
import { Text } from "./Text";

export const TextInfo = ({ title, id, text }) => {
  const [textInfo, setTextInfo] = useState({ totalText: false });

  const onShowText = () => {
    setTextInfo({
      totalText: !textInfo.totalText,
    });
  };

  return (
    <>
      <div className="textInfoDiv">
        <h4>Titulo: </h4>
        <p>{title}</p>
        <h4 className="infoId">ID: </h4>
        <p>{id}</p>
        {textInfo.totalText && (
          <Text text={text} />
        )}
      </div>
      <button onClick={() => onShowText()}>{textInfo.totalText ?  'Ocultar texto' : 'Ver texto completo' }</button>
    </>
  );
};
