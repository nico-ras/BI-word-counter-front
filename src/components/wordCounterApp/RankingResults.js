import React from "react";
import { WordTable } from "./WordTable";
import '../../styles/wordCounter/rankingResults.css'

export const RankingResults = ({ result }) => {
  const { id, title, ranking } = result;

  ranking && console.log(ranking.map((i) => i.position));
  return (
    <div>
      <hr />
      <h3 className="rankingTitle">Ranking de Palabras</h3>
      <div className="cardDiv">
        <h4>Titulo: </h4>
        <p>{title}</p>
        <p>ID: {id}</p>
        <h4>Resultados</h4>
        <table border="1">
          <tr>
            <th>Posici√≥n</th>
            <th>Palabra</th>
            <th>Ocurrencia</th>
            <th>Texto completo</th>
            <th>Editar</th>
          </tr>
          <tbody>
            {ranking && ranking.map((i) => <WordTable element={i} />)}
          </tbody>
        </table>
      </div>
    </div>
  );
};
