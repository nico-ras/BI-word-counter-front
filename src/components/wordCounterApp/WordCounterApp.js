import React, { useState } from "react";
import { RankingResults } from "./RankingResults";
import { TextForm } from "./TextForm";
import { useFetch } from "../../hooks/useFetch";
import '../../styles/wordCounter/wordCounterApp.css';

export const WordCounterApp = () => {
  const [formState, setFormState] = useState({ textId: "" });

  const { textId } = formState;

  const onInputChange = ({ target }) => {
    const { name, value } = target;
    setFormState({
      ...formState,
      [name]: value,
    });
  };

  const { data, isLoading, hasError, getFetch, } = useFetch( textId );


  return (
    <div className="app">
      <h1 className="wcaTitle">WordCounterApp</h1>

      <TextForm
        textId={textId}
        onInputChange={onInputChange}
        getFetch={getFetch}
      />
      { data && <RankingResults result={data} />}
      
    </div>
  );
};
