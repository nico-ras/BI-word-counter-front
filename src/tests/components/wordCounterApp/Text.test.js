const { render, screen } = require("@testing-library/react");
import { Text } from '../../../components/wordCounterApp/Text' 

describe('Pruebas en componente <Text />', () => { 

    const text = 'ES UN TEST';

    

    test('Debe ser como la captura', () => { 
        const { container } = render(<Text text={text} />);
        expect( container ).toMatchSnapshot();
     })


     test('Debe mostrar el texto literal "ES UN TEST"', () => { 
       
        render( <Text text={text} /> );
        // screen.debug()
        expect( screen.getByText(text) ).toBeTruthy;

        expect(screen.getByTestId('test-text').innerHTML).toBe(text)

      })

      
 })