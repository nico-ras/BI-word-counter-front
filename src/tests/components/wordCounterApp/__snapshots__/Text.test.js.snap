// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <Text /> Debe ser como la captura 1`] = `
<div>
  <div>
    <h4>
      Texto analizado: 
    </h4>
    <p
      data-testid="test-text"
    >
      ES UN TEST
    </p>
  </div>
</div>
`;
